### PR TITLE
Refactor internal context API

### PR DESCRIPTION
**Pull request by bloodearnest**
This refactors the previously multiple bits of request context into
a single object and API, that does not rely on werkzeug.

This cleans things up a lot, provides a much nicer context API to use,
and also allows us to get rid the the request_id module altogether.

It paves the way for a) dropping werkzeug as a required dependency and
b) using contextvars to store this new ContextData object

### CODE REVIEW

**Comment by bloodearnest** [comment]
This pull request **fixes 1 alert** when merging 11ecd489ae79d2731d4616829492bd84ce0c23ce into 9522eab0a822f6e9d8703d5e4ce0335203877513 - [view on LGTM.com](https://lgtm.com/projects/g/canonical-ols/talisker/rev/pr-67b5fac8c9ab241c33783840ba1ecb686ff01458)

**fixed alerts:**

* 1 for Module is imported with &#39;import&#39; and &#39;import from&#39;

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: tests/conftest.py [hunk 1]
```diff
@@ -65,7 +64,7 @@
 
 @pytest.yield_fixture(autouse=True)
 def clean_up(request, tmpdir, monkeypatch, config):
-    """Clean up all globals.
+    """Clean up all globals.    import pdb; pdb.set_trace()
```
Comment (tonysimpson) [inline tests/conftest.py]: Oops?

File: talisker/testing.py [hunk 2]
```diff
@@ -71,7 +72,8 @@ def temp_file():
 
 def clear_all():
     """Clear all talisker state."""
-    talisker.context.clear()  # talisker request_context
+    Context.clear()  # talisker request_context
```
Comment (tonysimpson) [inline talisker/testing.py]: why? - doesn't CONTEXTS.clear() below clear everything

File: tests/conftest.py (hunk 1 above)
Comment (bloodearnest) [reply tests/conftest.py]: oops indeed

File: talisker/testing.py (hunk 2 above)
Comment (bloodearnest) [reply talisker/testing.py]: It does, will fix

**Comment by bloodearnest** [comment]
This pull request **fixes 1 alert** when merging 8bdb3a6905d550e885be36160be6e695c9a7a816 into 9522eab0a822f6e9d8703d5e4ce0335203877513 - [view on LGTM.com](https://lgtm.com/projects/g/canonical-ols/talisker/rev/pr-d07e35b5bf56ecf498116cf275ada216eaa1c84f)

**fixed alerts:**

* 1 for Module is imported with &#39;import&#39; and &#39;import from&#39;

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**